xrootd/origin_test.go

```diff
@@ -134,4 +146,117 @@ func TestOrigin(t *testing.T) {
 	} else {
 		t.Fatalf("Unsucessful test: timeout when trying to send request to xrootd")
 	}
+	viper.Reset()
+}
+
+func TestS3OriginConfig(t *testing.T) {
+	viper.Reset()
+	tmpDir := t.TempDir()
+
+	// We need to start up a minio server, which is how we emulate S3. Open to better ways to do this!
+	minIOServerCmd := exec.Command("minio", "server", "--quiet", tmpDir)
+	minIOServerCmd.Env = []string{fmt.Sprintf("HOME=%s", tmpDir)}
+
+	// Create a few buffers to grab outputs
+	var outb, errb bytes.Buffer
+	minIOServerCmd.Stdout = &outb
+	minIOServerCmd.Stderr = &errb
+
+	err := minIOServerCmd.Start()
+	// Wait for the server to initialize. Hopefully this always happens in under 2 seconds!
```

Comment:
Can this do a small loop to test for server initialization like we do for the origin and xrootd?